framework/test/unit/modules/interoperability/mainchain/commands/initialize_message_recovery.spec.ts

```diff
@@ -151,6 +152,27 @@ describe('InitializeMessageRecoveryCommand', () => {
 			}).createCommandVerifyContext<MessageRecoveryInitializationParams>(command.schema);
 		});
 
+		it('should return error if channel data is invalid', async () => {
+			const transactionParams = {
+				...defaultParams,
+				channel: Buffer.alloc(0),
+			};
+			const encodedTransactionParams = codec.encode(
+				messageRecoveryInitializationParamsSchema,
+				transactionParams,
+			);
+			const context = createTransactionContext({
+				transaction: new Transaction({
+					...defaultTx,
+					command: command.name,
+					params: encodedTransactionParams,
+				}),
+				stateStore,
+			}).createCommandVerifyContext<MessageRecoveryInitializationParams>(command.schema);
+
+			await expect(command.verify(context)).rejects.toThrow('Invalid buffer length');
```

Comment:
Out of curiosity, shouldn't the `codec.encode` call already fail?

Reply:
For this test it does not because it's still an empty Buffer. Instead it fails at the schema validation. If it were an empty string it would fail on the codec.endcode.